enterprise_access/apps/content_assignments/content_metadata_api.py

```diff
@@ -17,22 +17,49 @@
 DEFAULT_STRFTIME_PATTERN = '%b %d, %Y'
 
 
+def _process_course_metadata(course_metadata, is_run_based_by_content_key, metadata_by_key):
+    """
+    Helper function to update the metadata_by_key dictionary with the course metadata based on
+    either a course run key (for run-based assignments) or a course key.
+    """
+    course_key = course_metadata.get('key')
+
+    # If a course run is assigned, update the metadata for the course run key. Otherwise,
+    # update the metadata for the course key.
+    for assignment_content_key, is_assigned_course_run in is_run_based_by_content_key.items():
+        if is_assigned_course_run:
+            metadata_by_key[assignment_content_key] = course_metadata
+        else:
+            # if not is_assigned_course_run, we can assume it's a (legacy) course-based assignment
+            metadata_by_key[course_key] = course_metadata
+
+
 def get_content_metadata_for_assignments(enterprise_catalog_uuid, assignments):
     """
     Fetches (from cache or enterprise-catalog API call) content metadata
     in bulk for the `content_keys` of the given assignments, provided
     such metadata is related to the given `enterprise_catalog_uuid`.
 
+    Note that the `content_keys` of the provided assignments may be
+    either course run keys or course keys.
+
```

Comment:
Nit: maybe also clarify that the metadata returned are all at the course-level, whether the key of the returned mapping is a run key or course key.